src/utils/generator.rs

```diff
@@ -0,0 +1,216 @@
+    Node {
+        uid: Uuid::new_v4().to_string(),
+        location: generate_location_near(location, radius),
+        forward_to: None,
+        status: status::Status::Ok,
+    }
+}
+
+/// Generate a random location anywhere on earth.
+///
+/// # Returns
+/// A random location anywhere on earth.
+pub fn generate_location() -> Location {
+    let mut rng = rand::thread_rng();
+    let latitude = OrderedFloat(rng.gen_range(-90.0..=90.0));
+    let longitude = OrderedFloat(rng.gen_range(-180.0..=180.0));
+    let altitude_meters = OrderedFloat(rng.gen_range(0.0..=10000.0));
+    Location {
+        latitude,
+        longitude,
+        altitude_meters,
+    }
+}
+
+/// Generate a random location near a given location and radius.
+///
+/// # Arguments
+/// * `location` - The location to generate a random location near.
+/// * `radius` - The radius in kilometers.
+///
+/// # Returns
+/// A random location near the given location and radius.
+pub fn generate_location_near(location: &Location, radius: f32) -> Location {
+    let mut rng = rand::thread_rng();
+    let (latitude, longitude) = gen_around_location(
+        &mut rng,
+        location.latitude.into_inner(),
+        location.longitude.into_inner(),
+        radius,
+    );
+
+    let altitude_meters = OrderedFloat(rng.gen_range(0.0..=10000.0));
+    Location {
+        latitude,
+        longitude,
+        altitude_meters,
+    }
+}
+
+/// Generate a random location within a radius.
+///
+/// Source: [Reddit](https://www.reddit.com/r/rust/comments/f08lqu/comment/fgsxeik/)
+///
+/// # Arguments
+/// * `rng` - The random number generator.
+/// * `latitude` - The latitude of the location.
+/// * `longitude` - The longitude of the location.
+/// * `radius` - The radius in kilometers.
+///
+/// # Returns
+/// A latitude and longitude pair.
+///
+/// # Notes
+/// @GoodluckH: This function sometimes output invalid coordinates. I'm not sure why.
+fn gen_around_location(
+    rng: &mut ThreadRng,
+    latitude: f32,
+    longitude: f32,
+    radius: f32,
+) -> (OrderedFloat<f32>, OrderedFloat<f32>) {
+    // Transform to cartesian coordinates
+    let x = (DEG_TO_RAD * longitude).cos();
+    let y = (DEG_TO_RAD * longitude).sin();
+    let z = (DEG_TO_RAD * latitude).sin();
+
+    // Generate random unit vector
+    let x1 = 2.0 * rng.gen::<f32>() - 1.0;
+    let y1 = 2.0 * rng.gen::<f32>() - 1.0;
+    let z1 = 2.0 * rng.gen::<f32>() - 1.0;
+    let len = (x1 * x1 + y1 * y1 + z1 * z1).sqrt();
+
+    // Generate random angle
+    let ang = 0.5 * (radius / 1000.0 * DEG_TO_RAD) * rng.gen::<f32>();
+    let ca = ang.cos();
+    let sa = ang.sin() / len;
+
+    // Create Quaternion components
+    let vec: Vector3<f32> = [x, y, z]; // Todo handle 0 case
+    let q: Quaternion<f32> = (ca, [sa * x1, sa * y1, sa * z1]);
+    let vec = quaternion::rotate_vector(q, vec);
+
+    let r_lon = RAD_TO_DEG * vec[1].atan2(vec[0]);
+    let r_lat = RAD_TO_DEG * vec[2].asin();
+    if r_lat.is_nan() {
```

Comment:
Want to check if r_lon is nan as well

Reply:
r_lon won't be nan because atan2 always returns a valid float whereas asin() might return nan if the value is outside of -1...1 range.